### PR TITLE
chore: Add (skipped) `ptytest` test that hangs on Intel Mac (and Windows)

### PR DESCRIPTION
**Pull request by mafredri**
This (draft) PR adds a test that breaks `ptytest` with unconsumed output on Intel Macs.

This is left open to inform / opportunity to investigate.

To reproduce: `go test ./pty/ptytest -v -run=TestPtytest/Do_not_hang -timeout=5s`

Here is the trace: [trace.txt](https://github.com/coder/coder/files/8741887/trace.txt)

// @coder/backend @Emyrk

### CODE REVIEW

**Comment by ammario** [comment]
What's the status of this PR?

**Comment by mafredri** [comment]
@ammario With the latest commits, this is now ready to be merged. Fixing this most likely requires a refactor (and decision on how to proceed) of the `ptytest` package.

I've opened https://github.com/coder/coder/issues/2122 to track it.

File: pty/ptytest/ptytest_test.go [hunk 1]
```diff
@@ -15,4 +21,41 @@ func TestPtytest(t *testing.T) {
...
+			{name: "1025 exceeds macOS buffer (must not hang)", output: strings.Repeat(".", 1025), isPlatformBug: true},
+			{name: "10241 large output", output: strings.Repeat(".", 10241), isPlatformBug: true}, // 1024 * 10 + 1
+		}
+		for _, tt := range tests {
+			tt := tt
+			// nolint:paralleltest // Avoid parallel test to more easily identify the issue.
+			t.Run(tt.name, func(t *testing.T) {
+				if tt.isPlatformBug && (runtime.GOOS == "darwin" || runtime.GOOS == "windows") {
+					t.Skip("This test hangs on macOS and Windows, see https://github.com/coder/coder/issues/2122")
+				}
```
Comment (Emyrk) [inline pty/ptytest/ptytest_test.go:43]: If this fails on mac, I'm good with merging it.

**Review by Emyrk — APPROVED** [review]

File: pty/ptytest/ptytest_test.go (hunk 1 above)
Comment (Emyrk) [reply pty/ptytest/ptytest_test.go:43]: Maybe push a commit without the skip to confirm it fails on ci?

Comment (mafredri) [reply pty/ptytest/ptytest_test.go:43]: > If this fails on mac, I'm good with merging it.

Yup, it fails on my Intel Mac (and CI), don't know about others 😄.

> Maybe push a commit without the skip to confirm it fails on ci?

I did in earlier commits: https://github.com/coder/coder/runs/6771839024?check_suite_focus=true

**Pull request merged**